### PR TITLE
implement dropdown dependency

### LINKED ISSUE #2: Increment 2
- [x] Dropdown dependency
 - Turn `UserProfileComponent.countries` / `UserProfileComponent.provinces` into a collection of `MyOption` interface aka `{ value: string; label: string }`
 - Map `ProvincesEnum` /  `ProvincesEnum` into a collection of `MyOption` : use for now the value o the enum as `Option.value` and `Option.label`. For i.e. `[ ... { label: CountriesEnum.CANADA, value: CountriesEnum.CANADA } ... ]`
 - Populate `UserProfileComponent.countries` accordingly
 - Populate `UserProfileComponent.provinces` accordingly to the `UserProfileComponent.userProfileForm.dropdownGroup.country` value change
- [x] Accessibility
 - Provide `<label>` to each field as per https://www.w3.org/WAI/tutorials/forms/labels/#associating-labels-explicitly

### PR DESCRIPTION
**Pull request by edwinyachoui-ibm**
Increment #2

- [DONE] Create MyOption Model
- [DONE] Map Province depending on country selected
- [DONE] Populate dropdown
- [DONE] Add labels

### CODE REVIEW

**Review by adrianobonfiglioibm — APPROVED** [review]

**Review by adrianobonfiglioibm — CHANGES_REQUESTED** [review]
So far so good. What about the label now ?

File: src/app/components/form-fields/dropdown/dropdown.component.html [hunk 1]
```diff
@@ -1,5 +1,7 @@
-<label>{{label}}</label>
+<label for="{{label}}">{{label}}</label>
   <select
+    name="{{label}}"
+    id="{{label}}"
```
Comment (adrianobonfiglioibm) [inline src/app/components/form-fields/dropdown/dropdown.component.html]: Let's assume the label is `OH MY GOD EDWIN !` .... is it still right ?

File: src/app/components/form-fields/input/input.component.html [hunk 2]
```diff
@@ -1,5 +1,7 @@
-<label>{{label}}</label>
-<input [type]="type" #input
+<label for="{{label}}">{{label}}</label>
+<input name="{{label}}"
```
Comment (adrianobonfiglioibm) [inline src/app/components/form-fields/input/input.component.html]: name property are pointless with Reactive Form ... and confusing

File: src/app/components/form-fields/input/input.component.html [hunk 3]
```diff
@@ -1,5 +1,7 @@
...
+       id="{{label}}"
+       [type]="type" #input
```
Comment (adrianobonfiglioibm) [inline src/app/components/form-fields/input/input.component.html]: `#input` why an id reference ?

**Review by adrianobonfiglioibm — CHANGES_REQUESTED** [review]

File: src/app/components/form-fields/dropdown/dropdown.component.html (hunk 1 above)
Comment (adrianobonfiglioibm) [reply src/app/components/form-fields/dropdown/dropdown.component.html]: This a lazy approach that won't work here. You better generate a true ID here. A quick searhc on stack overflow can save you a lot of time https://stackoverflow.com/questions/105034/how-to-create-a-guid-uuid
Good think is that, the id will be uniq regardless the amount of time your component will be used

Comment (edwinyachoui-ibm) [reply src/app/components/form-fields/dropdown/dropdown.component.html]: So I should use a different "@Input() name: string" property that's populated from the parent component ?

File: src/app/components/form-fields/input/input.component.html (hunk 3 above)
Comment (edwinyachoui-ibm) [reply src/app/components/form-fields/input/input.component.html]: by mistake

File: src/app/components/form-fields/dropdown/dropdown.component.html (hunk 1 above)
Comment (adrianobonfiglioibm) [reply src/app/components/form-fields/dropdown/dropdown.component.html]: `@input() name: string` is very confusing ...as is i feel like you want to feed a input name property like. Here we are just talking about `id`, that's it

**Review by adrianobonfiglioibm — APPROVED** [review]

**Pull request merged**